lib/utils/sbom-spdx.js

```diff
@@ -0,0 +1,175 @@
+
+const crypto = require('crypto')
+const normalizeData = require('normalize-package-data')
+const npa = require('npm-package-arg')
+const ssri = require('ssri')
+
+const SPDX_SCHEMA_VERSION = 'SPDX-2.3'
+const SPDX_DATA_LICENSE = 'CC0-1.0'
+const SPDX_IDENTIFER = 'SPDXRef-DOCUMENT'
+
+const NO_ASSERTION = 'NOASSERTION'
+
+const REL_DESCRIBES = 'DESCRIBES'
+const REL_PREREQ = 'HAS_PREREQUISITE'
+const REL_OPTIONAL = 'OPTIONAL_DEPENDENCY_OF'
+const REL_DEV = 'DEV_DEPENDENCY_OF'
+const REL_DEP = 'DEPENDS_ON'
```

Comment:
need to double check, but the direction might be wrong:
```suggestion
const REL_DEP = 'DEPENDENCY_OF'
```

At least the other relationships are in the other direction

Reply:
@maxhbr I was chatting with one of the SPDX contributors about this and the guidance I got was that the "directionality" of the relationship is implied by the `spdxElementId`/`relatedSpdxElement` fields not by the `relationshipType` -- many of the types don't have a supported inverse, so I just picked whichever one best conveys the type of the relationship w/o regard for the direction.

Perhaps, I should switch this one just to be consistent, but it was the one case where there was a supported type that actually matched the direction of the relationship.

My understanding is that this is going to get better in SPDX 3.

Reply:
Yes, the directionality is implied by the `spdxElementId`/`relatedSpdxElement` fields, in combination with the type of the relationship.

The following two relationships are equivalent
```
spdxElementId: PackageA
relatedSpdxElement: PackageB
relationshipType: DEPENDS_ON
```
and
```
spdxElementId: PackageB
relatedSpdxElement: PackageA
relationshipType: DEPENDENCY_OF
```

The direction of the type of relationship is still significant.

Reply:
In your case probably the `DEPENDS_ON` is the right choice, and the other mappings should be adopted.

Reply:
to clarify the comment above:

this does not resolve the issue. It is just a statement that, when choosing between `DEPENDS_ON` and `DEPENDENCY_OF` the first should probably be preferred (aligning with https://github.com/npm/cli/issues/6867#issuecomment-1749041505).

**But still the issue is valid and the `relationshipTypes` are inconsistently used here.**